extensions/amp-next-page/amp-next-page.md

```diff
@@ -25,237 +25,346 @@ limitations under the License.
+The maximum number of pages to load and show to the user. The maximum number should be less than the total amount of pages. After meeting the number, <amp-next-page> displays the footer recommendation box. The default is `Infinity`.
 
-### Example configuration
+### `deep-parsing` (optional)
 
-The following configuration will recommend two more documents for the user to
-read, and hides the header and footer elements from each child document.
+When specified, the `deep-parsing` attribute enables recursive parsing of inline JSON configurations from subsequently loaded documents. This only works if the loaded documents contain an `<amp-next-page>` tag.
 
-```json
-{
-  "pages": [
-    {
-      "image": "https://example.com/image1.jpg",
-      "title": "This article shows first",
-      "ampUrl": "https://example.com/article1.amp.html"
-    },
-    {
-      "image": "https://example.com/image2.jpg",
-      "title": "This article shows second",
-      "ampUrl": "https://example.com/article2.amp.html"
-    }
-  ],
-  "hideSelectors": [".header", ".footer"]
-}
-```
+This behavior is the default if the `<amp-next-page>` component only specifies an inline configuration. If the `<amp-next-page>` component loads the configuration remotely this behavior is disabled by default unless `deep-parsing` is set to `true`.
+
+### `xssi-prefix` (optional)
+
+When specified, this attribute enables `amp-next-page` to strip a prefix before parsing remotely hosted JSON. This can be useful for APIs that include [security prefixes](http://patorjk.com/blog/2013/02/05/crafty-tricks-for-avoiding-xssi/) like `)]}` to help prevent cross site scripting attacks.
+
+## Styling
+
+The `<amp-next-page>` component renders a default recommendation box and separator. You may style the appearance of these two elements as follows:
+
+### Style the default footer recommendation box
 
-## Substitutions
+You may add custom CSS styles to the default footer recommendation box. It exposes the following classes:
 
-The `amp-next-page` src allows all standard URL variable substitutions. See
-the [Substitutions Guide](../../spec/amp-var-substitutions.md) for more info.
+- `.amp-next-page-footer` for the parent container element
+- `.amp-next-page-footer-content` for the wrapping element
+- `.amp-next-page-footer-article` for an individual article which contains:
+- `.amp-next-page-footer-image` for the link image
+- `.amp-next-page-footer-text` for the link text
 
-For example:
+### Style the default page separator
+
+The default separator (shown between article recommendations) is a simple gray horizontal rule, although this can be customized through CSS by applying styles to the `.amp-next-page-separator` class, templating the separator allows for more flexibility in styling and allows the integration of the title, image and URL of the upcoming article into the separator's appearance.
```

Comment:
```suggestion
Each document loads with a full-width hairline to separate it from the previous page. The default color is gray. Customize it through CSS using the `.amp-next-page-separator` class. 

```css
.amp-next-page-separator {
     color: red
     width: 5px
}
```
```

Reply:
Hmm this makes it look like the only thing you can customize here is color, I changed the wording a bit

Reply:
Perfect, thank you!